### ISSUE TITLE
Ensure `repoFolderName`s are unique

### ISSUE THREAD

**Issue by bluein-green** [reposense/RepoSense#807]
If we have 2 (local) repos with paths:
- .... /dir1/name_of_repo
- .... /dir2/name_of_repo

`repoFolderName` for the 2 `RepoConfiguration`s will be "name_of_repo". Thus for both repos, the folder that contains the cloned data will be named `/repos/name_of_repo`.

Consequently, when repo 1 has been cloned, and RepoSense attempts to clone repo 2 and analyse repo 1 in parallel, the analysis of repo 1 would try to refer to `/repos/name_of_repo`, which could have been deleted (as part of cloning repo 2) or could contain cloned data of repo 2.

On my system, running Reposense with 2 such local repos together results in analysis failure for both repos, while doing so 1 repo at a time generates the reports successfully.

I believe this is somewhat related to #228.

**Suggestion**
Append `location.hashCode()` to the `repoFolderName` when constructing each `RepoConfiguration`. Using the hashCode provides a deterministic and (mostly) unique way of identifying each folder as compared to appending a count value or such.

Or use a name-based UUID derived from the full `location` of the repo for the `repoFolderName`

**Comment by damithc** [comment]
Let's put on hold as supporting local repos is not a priority.
